src/object.d

```diff
@@ -1751,6 +1751,59 @@ unittest
 }
 
 
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Example:
+ * ---
+ * class MeaCulpa: Exception
+ * {
+ *     mixin StandardExceptionConstructor;
+ * }
+ * assertThrown!MeaCulpa(enforce!MeaCulpa(false));
+ *
+ * try throw new MeaCulpa;
+ * catch (MeaCulpa e)
+ * {
+ *     assert(e.file == __FILE__);
+ *     assert(e.line == __LINE__ - 4);
+ * }
+ * ---
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructor()
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__)
+    {
+        super(msg, file, line);
+    }
```

Comment:
I still think that this is a waste of time, because you the constructors that get mixed in can't be documented, but if you're going to do do this, the correct constructors would be

    this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null) @safe pure nothrow
    {
        super(msg, file, line, next);
    }

    this(string msg, Throwable next, string file = __FILE__, size_t line = __LINE__) @safe pure nothrow
    {
        super(msg, file, line, next);
    }